workflow-controller/controller-service/src/main/java/io/hyscale/controller/util/LoggerUtility.java

```diff
@@ -121,6 +124,70 @@ public void deploymentLogs(WorkflowContext context) throws HyscaleException {
             WorkflowLogger.footer();
         }
     }
+    
+    /**
+     * implementation :
+     * <b>
+     * 1. Replica name provided by user, validate and return replica name
+     * 2. Replica name not provided by user
+     *      a. If single replica, return replica name
+     *      b. else prompt user to provide replica name
+     * </b>
+     * @param authConfig
+     * @param appName
+     * @param serviceName
+     * @param replicaName
+     * @param namespace
+     * @return replica name
+     * @throws HyscaleException
+     */
+    private String getPodName(AuthConfig authConfig, String appName, String serviceName, String replicaName,
+            String namespace) throws HyscaleException {
+        List<V1Pod> existingPods = null;
+        try {
+            existingPods = getExistingPods(authConfig, appName, serviceName, namespace);
+        } catch (HyscaleException ex) {
+            WorkflowLogger.error(ControllerActivity.FAILED_TO_STREAM_SERVICE_LOGS, ex.getMessage());
+            throw ex;
+        }
+        if (existingPods == null || existingPods.isEmpty()) {
+            WorkflowLogger.error(ControllerActivity.SERVICE_NOT_CREATED);
+            WorkflowLogger.error(ControllerActivity.CHECK_SERVICE_STATUS);
+            WorkflowLogger.footer();
+            return null;
+        }
+        
+        if (replicaName != null) {
+            boolean isReplicaNameValid = existingPods.stream().anyMatch(each -> each.getMetadata().getName().equals(replicaName));
+            if (!isReplicaNameValid) {
+                StringBuilder pods = new StringBuilder();
+                existingPods.forEach(each -> pods.append(each.getMetadata().getName() + ", "));
+                WorkflowLogger.error(ControllerActivity.REPLICA_DOES_NOT_EXIT, replicaName,
+                        HyscaleStringUtil.removeSuffixStr(pods, ", "));
+                WorkflowLogger.footer();
+                HyscaleException ex = new HyscaleException(DeployerErrorCodes.REPLICA_DOES_NOT_EXIT, replicaName);
+                throw ex;
+            }
+            return replicaName;
+        }
+        
+        if (existingPods.size() == 1) {
+            return existingPods.get(0).getMetadata().getName();
+        }
+        
+        ApiClient apiClient = clientProvider.get((K8sAuthorisation) authConfig);
+        existingPods = K8sDeployerUtil.filterPods(apiClient, appName, serviceName, namespace, existingPods);
+        return getPodFromUser(authConfig, appName, serviceName, namespace, existingPods);
```

Comment:
if replica name is present
     validate and stream logs to user
else 
    read the replica from end user
    validate the input 
    stream the logs .